yarn-project/end-to-end/src/e2e_key_registry.test.ts

```diff
@@ -0,0 +1,252 @@
+        .send()
+        .wait();
+
+      const rawLogs = await pxe.getUnencryptedLogs({ txHash });
+      expect(Fr.fromBuffer(rawLogs.logs[0].log.data)).toEqual(Fr.ZERO);
+    });
+
+    it('checks our registry contract from state vars contract and finds the address and associated nullifier public key after a delay', async () => {
+      await delay(5);
+
+      const { txHash } = await stateVarsContract.methods
+        .test_shared_mutable_private_getter_for_registry_contract(keyRegistry.address, 1, accountAddedToRegistry)
+        .send()
+        .wait();
+
+      const rawLogs = await pxe.getUnencryptedLogs({ txHash });
+
+      expect(Fr.fromBuffer(rawLogs.logs[0].log.data)).toEqual(new Fr(12));
+    });
+  });
+
+  describe('key rotation flow', () => {
+    it('we rotate the nullifier key', async () => {
+      // This changes
+      const newMasterNullifierPublicKey = new Fr(910);
+
+      await keyRegistry
+        .withWallet(wallets[0])
+        .methods.rotate_keys(wallets[0].getAddress(), newMasterNullifierPublicKey)
+        .send()
+        .wait();
+    });
+
+    it("checks our registry contract from state vars contract and finds our old public key because the key rotation  hasn't been applied yet", async () => {
+      const { txHash } = await stateVarsContract.methods
+        .test_shared_mutable_private_getter_for_registry_contract(keyRegistry.address, 1, wallets[0].getAddress())
+        .send()
+        .wait();
+
+      const rawLogs = await pxe.getUnencryptedLogs({ txHash });
+      expect(Fr.fromBuffer(rawLogs.logs[0].log.data)).toEqual(new Fr(0));
+    });
+
+    it('checks our registry contract from state vars contract and finds the new nullifier public key that has been rotated', async () => {
+      await delay(5);
+
+      const { txHash } = await stateVarsContract.methods
+        .test_shared_mutable_private_getter_for_registry_contract(keyRegistry.address, 1, wallets[0].getAddress())
+        .send()
+        .wait();
+
+      const rawLogs = await pxe.getUnencryptedLogs({ txHash });
+
+      expect(Fr.fromBuffer(rawLogs.logs[0].log.data)).toEqual(new Fr(910));
+    });
+  });
+
+  describe('key rotation flow with authwit', () => {
+    it('wallet 0 lets wallet 1 call rotate_keys on his behalf with a pre-defined new public key', async () => {
+      // This changes
+      const newMasterNullifierPublicKey = new Fr(420);
+
+      const action = keyRegistry
+        .withWallet(wallets[1])
+        .methods.rotate_keys(wallets[0].getAddress(), newMasterNullifierPublicKey);
+
+      await wallets[0]
+        .setPublicAuthWit({ caller: wallets[1].getCompleteAddress().address, action }, true)
+        .send()
+        .wait();
+
+      await action.send().wait();
+    });
+
+    it("checks our registry contract from state vars contract and finds our old public key because the key rotation  hasn't been applied yet", async () => {
```

Comment:
```suggestion
    it("checks our registry contract from state vars contract and finds our old public key because the key rotation hasn't been applied yet", async () => {
```
My autisms got triggered